hedera-node/hedera-mono-service/src/main/java/com/hedera/node/app/service/mono/store/contracts/precompile/HTSPrecompiledContract.java

```diff
@@ -819,6 +848,14 @@ private Precompile checkFungible(final boolean isFungible, final Supplier<Precom
         }
     }
 
+    private Precompile checkHRC(final boolean validToken, final Supplier<Precompile> precompileSupplier) {
```

Comment:
nit: since this is targeted for Token vs other entities
```suggestion
    private Precompile checkHRCToken(final boolean validToken, final Supplier<Precompile> precompileSupplier) {
```